src/components/accounts/recovery_setup/SetupRecoveryMethod.js

```diff
@@ -101,18 +132,33 @@ class SetupRecoveryMethod extends Component {
     }
 
     handleSendCode = () => {
-        const  { accountId, initializeRecoveryMethod } = this.props;
+        const  { accountId, initializeRecoveryMethod, isNew } = this.props;
 
-        initializeRecoveryMethod(accountId, this.method);
+        initializeRecoveryMethod(accountId, this.method, isNew);
         this.setState({ success: true })
         
     }
 
     handleSetupRecoveryMethod = async (securityCode) => {
-        const  { accountId, setupRecoveryMessage, redirectToApp } = this.props;
+        const  {
+            accountId, setupRecoveryMessage, redirectToApp, history,
+            isNew, fundingContract, fundingKey, refreshAccount
+        } = this.props;
+
+        try {
+            await setupRecoveryMessage(accountId, this.method, securityCode, isNew, fundingContract, fundingKey)
+            await refreshAccount()
+        } catch(e) {
+            console.log(e);
```

Comment:
why is error ignored here? If there is a good reason – it should be a comment in code, as everyone reading it will be asking this question.